Add LSP configuration to use sqls for SQL

v2okimochi opened the pull request:
I added LSP configuration for [sqls](https://github.com/lighttiger2505/sqls).

To use sqls, it need following steps:
- Get sqls through golang ( `go get`) according to sqls document
- Put workspace configuration to connect to DB

[There are some configuration methods](https://github.com/lighttiger2505/sqls#configuration-methods) to use sqls features, and this PR has one of method that `workspace/configuration`.

We can use this method through creating `config.json` under `.sqls/` at workspace root directory.

Sample `config.json` is following (cf. [workspace configuration sample by sqls official](https://github.com/lighttiger2505/sqls#workspace-configuration-sample))

```json:workspace-root-dir/.sqls/config.json
{
  "sqls": {
    "connections": [
      {
        "driver": "mysql",
        "dataSourceName": "user1:password1@tcp(localhost:3306)/sample_db"
      }
    ]
  }
}
```

If anyone have more great way than this PR, I would like to agree it.

yyoncho commented on lsp-sqls.el:
```diff
@@ -0,0 +1,67 @@
+  "LSP support for SQL, using sqls"
+  :group 'lsp-mode
+  :link '(url-link "https://github.com/lighttiger2505/sqls")
+  :package-version `(lsp-mode . "6.3"))
```
can you change all these to 6.4?

yyoncho commented on lsp-sqls.el:
```diff
@@ -0,0 +1,67 @@
+
+(defun lsp-sqls-workspace-config-json ()
+  "Workspace configuration json at workspace root directory."
+  (let ((config-json-path (-> (lsp-workspace-root)
```
can you also add a defcustom which can hold this structure as well so you will be able to pass the configuration with elisp as well?

kiennq commented on lsp-sqls.el:
```diff
@@ -0,0 +1,67 @@
+                              (concat "/.sqls/config.json"))))
+    (if (file-exists-p config-json-path)
+        (json-read-file config-json-path)
+      "{ \"sqls\": {} }")))
```
Shouldn't this a JSON object instead of string?
Also `json-read-file` doesn't benefit from native json. Perhaps you can create a `lsp--read-json-file` that looks like `json-read-file` and use `lsp--read-json` to parse json file instead?

ericdallo commented on mkdocs.yml:
```diff
@@ -82,6 +82,7 @@ nav:
     - Vue: page/lsp-vetur.md
     - XML: page/lsp-xml.md
     - YAML: page/lsp-yaml.md
+    - SQL (sqls): page/lsp-sqls.md
```
Could you add that in alphabetical order?

ericdallo commented on lsp-sqls.el:
```diff
@@ -0,0 +1,76 @@
+  :type 'file
+  :package-version `(lsp-mode . "6.4"))
+
+(defcustom lsp-sqls-workspace-config-path "workspace"
```
Instead of allowing any string, you can use as a choice like [here](https://github.com/emacs-lsp/lsp-mode/blob/31fe3c077fbb61d42e48108c2d02ca561f57c7be/lsp-vetur.el#L57-L59)

v2okimochi commented:
@yyoncho @kiennq Thanks for your reviews. I modified code.
I added defcustom `lsp-sqls-workspace-config-path` to choice path to workspace configuration directory.
Possible values are:
- "workspace": {current workspace}/.sqls/config.json
- "root": {root directory of workspace}/.sqls/config.json

I would like your re-review and any indicates if I have made mistake your intend.

Best Regards,
v2okimochi

yyoncho commented:
@v2okimochi thank you for contributing to `lsp-mode` and congrats for your first `lsp-mode` PR! :tada:

This is valuable contribution!

:purple_heart:
